### PR TITLE
[Instrumentation.SqlClient] Implements database metric `db.client.operation.duration` (part 2/netfx)

### PR DESCRIPTION
**Pull request by matt-hensley**
## Changes

Implements proposed `db.client.operation.duration` metric for SqlClient.

Adds implementation for .NET Framework.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlEventSourceTests.netfx.cs [hunk 1]
```diff
@@ -323,6 +356,41 @@ private static void VerifyActivityData(
     }
 
 #pragma warning disable SA1201 // Elements should appear in the correct order
+    public static IEnumerable<object[]> EventSourceFakeTestCases()
```
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlEventSourceTests.netfx.cs:359]: You're getting a build error here

> 'public' members should come before 'private' members

File: src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs [hunk 2]
```diff
@@ -29,6 +29,19 @@ internal sealed class SqlEventSourceListener : EventListener
     internal const int BeginExecuteEventId = 1;
     internal const int EndExecuteEventId = 2;
 
+    private static readonly string[] SharedTagNames =
```
[the remark this answers is not in the document]
Comment (matt-hensley) [reply src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlEventSourceListener.netfx.cs:32]: Yep, went for the expedient route to simplify reviewing. Easy enough to extract, will simplify testing too.

File: test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlActivitySourceHelperTests.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+using OpenTelemetry.Instrumentation.SqlClient.Implementation;
+using Xunit;
+
+namespace OpenTelemetry.Instrumentation.SqlClient.Tests;
+
+public class SqlActivitySourceHelperTests
+{
+    [Fact]
+    public void ShouldCalculateDuration()
```
Comment (alanwest) [inline test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlActivitySourceHelperTests.cs:12]: Failing test on ubuntu

```text
  Failed OpenTelemetry.Instrumentation.SqlClient.Tests.SqlActivitySourceHelperTests.ShouldCalculateDuration [24 ms]
  Error Message:
   Assert.Equal() Failure: Values differ
Expected: 10
Actual:   0.10000000000000001
  Stack Trace:
     at OpenTelemetry.Instrumentation.SqlClient.Tests.SqlActivitySourceHelperTests.ShouldCalculateDuration() in /home/runner/work/opentelemetry-dotnet-contrib/opentelemetry-dotnet-contrib/test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlActivitySourceHelperTests.cs:line 16
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

Comment (matt-hensley) [reply test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlActivitySourceHelperTests.cs]: Dropped the test for now, was just a nice bonus but not that valuable.

**Review by alanwest — APPROVED** [review]

**Pull request merged**